src/VisualStudio/Core/Def/Implementation/TaskList/ExternalErrorDiagnosticUpdateSource.cs

```diff
@@ -127,6 +127,14 @@ internal sealed class ExternalErrorDiagnosticUpdateSource : IDiagnosticUpdateSou
         /// </summary>
         public bool IsInProgress => BuildInProgressState != null;
 
+        public void Dispose()
+        {
+            lock (_gate)
+            {
+                _activeCancellationSeriesDoNotAccessDirectly.Dispose();
+            }
```

Comment:
interesting.  is this lock necessary?  what difference woudl we have if we didn't have this?

Reply:
It's probably not necessary, but it also doesn't hurt. I kept it to avoid questions about why there appears to be an unguarded access to a state field.